app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/RegisterWithEoriAndIdController.scala

```diff
@@ -297,8 +297,10 @@ class RegisterWithEoriAndIdController @Inject() (
     id: String
   )(implicit request: Request[AnyContent], loggedInUser: LoggedInUserWithEnrolments, hc: HeaderCarrier) =
     subscriptionStatusService.getStatus(idType, id).flatMap {
-      case NewSubscription | SubscriptionRejected =>
+      case NewSubscription =>
         onSuccessfulSubscriptionStatusSubscribe(service, journey)
+      case SubscriptionRejected =>
```

Comment:
Rejected Trader can try again put the correct details, So  he can  be successful, This solution is inline by CDIO architect, Trader will know the rejected reason, in this case, he won't be successful  if we remove this option

Reply:
We noticed that we're handling this SubscriptionRejected as a successful and we have rejected endpoint which is not used anywhere. We thought that this scenario where we received the SubscriptionRejected we should use this screen.
Additionally with this approach for SubscriptionRejected we're doing a SUB02 call after, so we wanted to avoid unnecessary call (keeping in mind that the subscription is rejected).

Based on what you're saying this is working as expected, so we will keep this implementation.

Thanks for explanation @shaileshp0110